### PR TITLE
Displaying data in cell

### PR DESCRIPTION
**Pull request by inetamagone**
Displaying data in cell

### CODE REVIEW

File: MI-Xib_cell/Controllers/ViewController.swift [hunk 1]
```diff
@@ -42,8 +41,8 @@ extension ViewController: UITableViewDelegate, UITableViewDataSource {
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         guard let cell = tableView.dequeueReusableCell(withIdentifier: CustomTableViewCell.reuseId, for: indexPath) as? CustomTableViewCell else {return .init()}
-        
-        cell.configure(labelList: labelData)
+        cell.roleLabel.text = labelData[indexPath.row].role
+        cell.nameLabel.text = labelData[indexPath.row].participantName
```
Comment (eugenemokhan) [inline MI-Xib_cell/Controllers/ViewController.swift]: pls, use configure function to set values

File: MI-Xib_cell/Views/CustomTableViewCell.swift [hunk 2]
```diff
@@ -36,21 +34,17 @@ class CustomTableViewCell: UITableViewCell {
...
             commonView.trailingAnchor.constraint(equalTo: trailingAnchor)
         ])
     }
-
+    
     override func setSelected(_ selected: Bool, animated: Bool) {
         super.setSelected(selected, animated: animated)
     }
     
-    func configure (labelList: [LabelData]) {
```
Comment (eugenemokhan) [inline MI-Xib_cell/Views/CustomTableViewCell.swift:52]: think there your should get an array, but should take labelData

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

File: MI-Xib_cell/Controllers/ViewController.swift [hunk 3]
```diff
@@ -42,17 +42,17 @@ extension ViewController: UITableViewDelegate, UITableViewDataSource {
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         guard let cell = tableView.dequeueReusableCell(withIdentifier: CustomTableViewCell.reuseId, for: indexPath) as? CustomTableViewCell else {return .init()}
-        
-        cell.configure(labelList: labelData)
+        let data = labelData[indexPath.row]
```
Comment (eugenemokhan) [inline MI-Xib_cell/Controllers/ViewController.swift]: send this data to the cell pls
cell.configure(data: data)

File: MI-Xib_cell/Controllers/ViewController.swift [hunk 4]
```diff
@@ -42,17 +42,17 @@ extension ViewController: UITableViewDelegate, UITableViewDataSource {
...
+        cell.roleLabel.text = data.role
+        cell.nameLabel.text = data.participantName
+//        cell.configure(labelList: labelData)
```
Comment (eugenemokhan) [inline MI-Xib_cell/Controllers/ViewController.swift]: at cell at method configure set this properties

func configure(data: LabelData){
roleLabel.text = data.role
}

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

File: MI-Xib_cell/Controllers/ViewController.swift [hunk 5]
```diff
@@ -43,16 +41,13 @@ extension ViewController: UITableViewDelegate, UITableViewDataSource {
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         guard let cell = tableView.dequeueReusableCell(withIdentifier: CustomTableViewCell.reuseId, for: indexPath) as? CustomTableViewCell else {return .init()}
         
-        cell.configure(labelList: labelData)
+        let indexOfRow = labelData[indexPath.row]
+        cell.configure(labelList: labelData, indexOfRow: indexOfRow)
```
Comment (eugenemokhan) [inline MI-Xib_cell/Controllers/ViewController.swift]: Why you send labelData?

File: MI-Xib_cell/Views/CustomTableViewCell.swift [hunk 6]
```diff
@@ -8,23 +8,23 @@
...
-
+    
     @IBOutlet weak var commonView: UIView!
     @IBOutlet weak var roleLabel: UILabel!
     @IBOutlet weak var nameLabel: UILabel!
     
     static var reuseId: String = "customCell"
-    private var labelData: [LabelData] = []
+    
+    var labelData: [LabelData] = []
```
Comment (eugenemokhan) [inline MI-Xib_cell/Views/CustomTableViewCell.swift]: Why you take there [LabelData] not LabelData?

**Review by eugenemokhan — CHANGES_REQUESTED** [review]

File: MI-Xib_cell/Controllers/ViewController.swift (hunk 5 above)
Comment (inetamagone) [reply MI-Xib_cell/Controllers/ViewController.swift]: I send labelData to get label text data from ViewController to cell file and execute the configure function.

File: MI-Xib_cell/Views/CustomTableViewCell.swift (hunk 6 above)
Comment (inetamagone) [reply MI-Xib_cell/Views/CustomTableViewCell.swift]: I will try like you say with LabelData?

File: MI-Xib_cell/Controllers/ViewController.swift (hunk 5 above)
Comment (eugenemokhan) [reply MI-Xib_cell/Controllers/ViewController.swift]: what is difference between indexOfRow?

File: MI-Xib_cell/Views/CustomTableViewCell.swift (hunk 6 above)
Comment (eugenemokhan) [reply MI-Xib_cell/Views/CustomTableViewCell.swift]: you have an model LabelData, if you will get the variable with name labelsData i will say to use, that you should send labelData, but you have an name labelData, idk how to say to you, that you should send single object of you array of data

File: MI-Xib_cell/Controllers/ViewController.swift (hunk 5 above)
Comment (inetamagone) [reply MI-Xib_cell/Controllers/ViewController.swift]: If I do not take the indexOfRow, I can not put the label names in the right places

File: MI-Xib_cell/Views/CustomTableViewCell.swift (hunk 6 above)
Comment (inetamagone) [reply MI-Xib_cell/Views/CustomTableViewCell.swift]: I took the array in cell file and changed the function.

File: MI-Xib_cell/Views/CustomTableViewCell.swift [hunk 7]
```diff
@@ -35,22 +34,19 @@ class CustomTableViewCell: UITableViewCell {
...
     
-    func configure (labelList: [LabelData]) {
-        self.labelData = labelList
+    func configure(indexOfRow: LabelData) {
+        var roleValue = ""
+        var nameValue = ""
+        
+        roleValue = indexOfRow.role
+        nameValue = indexOfRow.participantName
+        self.roleLabel.text = roleValue
```
Comment (eugenemokhan) [inline MI-Xib_cell/Views/CustomTableViewCell.swift:49]: why don't use self.roleLabel.text = indexOfRow.role?

**Review by eugenemokhan — APPROVED** [review]

**Pull request merged**